session_rpcserver.go

```diff
@@ -100,12 +108,52 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
 	}
 	if sess.Expiry.Before(time.Now()) {
 		log.Debugf("Not resuming session %x with expiry %s",
-			sess.LocalPublicKey.SerializeCompressed(), sess.Expiry)
+			localPubKey, sess.Expiry)
+
+		if err := s.db.RevokeSession(pubKey); err != nil {
+			return fmt.Errorf("error revoking session: %v", err)
+		}
+
 		return nil
 	}
 
 	authData := []byte("Authorization: Basic " + s.basicAuth)
-	return s.sessionServer.StartSession(sess, authData)
+	if err = s.sessionServer.StartSession(sess, authData); err != nil {
```

Comment:
This is the only place where we call `StartSession` so maybe it's a bit cleaner to return the `sessionClosedSub` here? See previous comment.

Reply:
great idea!